src/modules/Generator.jl

```diff
@@ -0,0 +1,86 @@
+"""
+Provides the functions related to generating documentation stubs.
+"""
+module Generator
+
+"""
+    savefile(f, root, filename)
+
+Attempts to save a file at `\$(root)/\$(filename)`. `f` will be called with file
+stream (see [`open`](http://docs.julialang.org/en/latest/stdlib/io-network/#Base.open)).
+"""
+function savefile(f, root, filename)
+    filepath = joinpath(root, filename)
+    if ispath(filepath) error("$(filepath) already exists") end
+    open(f,filepath,"w")
+end
+
+"""
+Contents of the default `make.jl` file.
+"""
+function make(pkgname)
+    """
+    using Documenter
+    using $(pkgname)
+
+    makedocs(
+        modules = [$(pkgname)]
+    )
+
+    #deploydocs()
+    """
+end
+
+"""
+Contents of the default `.gitignore` file.
+"""
+function gitignore()
+    """
+    build/
+    site/
+    """
+end
+
+"""
+Contents of the default `mkdocs.yml` file.
+"""
+function mkdocs(pkgname)
+    """
+    site_name:        $(pkgname).jl
+    #repo_url:         https://github.com/USER_NAME/PACKAGE_NAME.jl
+    #site_description: Description...
+    #site_author:      USER_NAME
```

Comment:
Yeah, would be nice, but I am not sure where to get the information from (automatically). Of course, `generate` could take keywords, but I think it is easier for the user to change this stuff manually later.

I was thinking that we could check the repo's `origin` ref and take the URL from there, but I didn't look into that yet.

The `mkdocs()` could nevertheless take these things as arguments though, even if they do not get used right now..